### PR TITLE
feat(frontend): task explanations bb-340

### PR DESCRIPTION
**Pull request by Mikoyzskie**
- Add explain task button
- Generate explanation from openai
-

### CODE REVIEW

**Comment by Mikoyzskie** [comment]
image

File: apps/frontend/src/modules/chat/slices/chat.slice.ts [hunk 1]
```diff
@@ -107,6 +110,55 @@ const { actions, name, reducer } = createSlice({
...
+			})
+			.addCase(explainTasksSuggestions.rejected, (state) => {
+				state.dataStatus = DataStatus.REJECTED;
+			})
+			.addCase(explainTasksSuggestions.fulfilled, (state, action) => {
+				state.dataStatus = DataStatus.FULFILLED;
+				const newMessages = action.payload.messages;
+				const taskExplanationsMessagePayload: TaskMessage[] = [];
+
+				for (const message of newMessages) {
```
Comment (fshabanov) [inline apps/frontend/src/modules/chat/slices/chat.slice.ts:125]: We now have this logic in a helper, in shared

**Review by fshabanov — CHANGES_REQUESTED** [review]
After displaying the explanations, we should have only 2 buttons: Add these tasks and I don't like these tasks. We shouldn't have explain button again

**Review by ihorLysak — COMMENTED** [review]
Shouldn't be too hard to resolve conflicts for this one.

File: apps/frontend/src/pages/chat/libs/components/buttons-controller/libs/components/suggestions-manipulation-options/suggestions-manipulation-options.tsx [hunk 2]
```diff
@@ -13,8 +13,15 @@ import {
 } from "./libs/enums/enums.js";
 import styles from "./styles.module.css";
 
-const SuggestionsManipulationOptions: React.FC = () => {
+type Properties = {
+	isExplained?: boolean;
+};
+
+const SuggestionsManipulationOptions: React.FC<Properties> = ({
+	isExplained,
```
Comment (fshabanov) [inline apps/frontend/src/pages/chat/libs/components/buttons-controller/libs/components/suggestions-manipulation-options/suggestions-manipulation-options.tsx:21]:
```suggestion
	isExplained = false,
```

**Review by fshabanov — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**